src/cloud-api-adaptor/podvm/hack/smoke_test.sh

```diff
@@ -0,0 +1,179 @@
+SCRIPTDIR=$(dirname "$(realpath "$0")")
+
+
+cleanup() {
+	# Cleanup (only when DESTRUCTIVE!=1)
+	if [ "${DESTRUCTIVE}" -ne 1 ]; then
+		set +e
+		popd
+		[ -n "${SOCAT_PID}" ] && kill "${SOCAT_PID}"
+		sudo virsh destroy smoketest
+		rm -Rf "${WORKDIR}"
+	fi
+}
+
+trap 'cleanup' EXIT ERR
+
+# Ensure we have kata-agent-ctl
+KATACTL=$(which --skip-alias kata-agent-ctl 2>/dev/null || true)
+if [ -z "${KATACTL}" ]; then
+	if [ -e kata-agent-ctl ]; then
+		KATACTL=$(realpath kata-agent-ctl)
+		echo "::debug:: Using kata-agent-ctl from this directory"
+	fi
+else
+	echo "::debug:: Using kata-agent-ctl from PATH ${KATACTL}"
+fi
+pushd "$WORKDIR"
+if [ -z "$KATACTL" ]; then
+	KATA_REF=$(yq -e '.oci.kata-containers.reference' ${SCRIPTDIR}/../../versions.yaml)
+	KATA_REG=$(yq -e '.oci.kata-containers.registry' ${SCRIPTDIR}/../../versions.yaml)
+	echo "::debug:: Pulling kata-ctl from ${KATA_REG}/agent-ctl:${KATA_REF}-x86_64"
+	oras pull "${KATA_REG}/agent-ctl:${KATA_REF}-x86_64"
+	tar -xJvf kata-static-agent-ctl.tar.xz ./opt/kata/bin/kata-agent-ctl --transform='s/opt\/kata\/bin\/kata-agent-ctl/kata-agent-ctl/'
+	rm kata-static-agent-ctl.tar.xz
+	KATACTL=$(realpath kata-agent-ctl)
+fi
+
+# Create cloud-init iso
+echo "::debug:: Preparing cloud-init iso"
+mkdir cloud-init
+touch cloud-init/meta-data
+cat <<EOF > cloud-init/user-data
+#cloud-config
+
+write_files:
+- path: /run/peerpod/daemon.json
+  content: |
+    {
+        "pod-network": {
+            "podip": "10.244.1.21/24",
+            "pod-hw-addr": "32:b9:59:6b:f0:d5",
+            "interface": "eth0",
+            "worker-node-ip": "10.224.0.5/16",
+            "tunnel-type": "vxlan",
+            "routes": [
+                {
+                    "dst": "0.0.0.0/0",
+                    "gw": "10.244.1.1",
+                    "dev": "eth0",
+                    "protocol": "boot"
+                },
+                {
+                    "dst": "10.244.1.0/24",
+                    "gw": "",
+                    "dev": "eth0",
+                    "protocol": "kernel",
+                    "scope": "link"
+                }
+            ],
+            "neighbors": null,
+            "mtu": 1500,
+            "index": 2,
+            "vxlan-port": 8472,
+            "vxlan-id": 555002,
+            "dedicated": false
+        },
+        "pod-namespace": "default",
+        "pod-name": "smoketest"
+    }
+EOF
+genisoimage -output cloud-init.iso -volid cidata -joliet -rock cloud-init/user-data cloud-init/meta-data
+
+# Move files to libvirt-accessible location
+echo "::debug:: Moving files to libvirt-accessible location"
+IMAGE=$(realpath "./podvm.${FMT}")
+if [ "${DESTRUCTIVE}" -eq 1 ]; then
+	mv "${IMG}" "${IMAGE}"
+else
+	cp "${IMG}" "${IMAGE}"
+fi
+sudo chown -R libvirt-qemu "${WORKDIR}" || true
+sudo chmod +x "${WORKDIR}"
+
+# Start the VM
+echo "::debug:: Starting VM"
+# TODO: Add AAVMF for arm
+[ -e "/usr/share/OVMF/OVMF_CODE.fd" ] && OVMF="/usr/share/OVMF/OVMF_CODE.fd"
```

Comment:
note, this will not work with all OVMF builds, e.g. not with the ones bundled in ubuntu 24.04. the best idea is to download the OVMFs. from a static resource. See the azure podvm-build workflow where we do this for creating the measurements file.

Maybe we don't need to use libvirt for the smoke test? I started with it, because of the libvirt provider, but it's maybe too invasive for local usage, plain qemu could do the same probably.

Reply:
Well I thought about that too but we'd have to query for IP so I kept using libvirt.

Reply:
what if we use qemu's hostfwd, so the host port can be `HOST_PORT="localhost:15150"`?
```
-netdev type=user,id=user-net,hostfwd=tcp::15150-:15150 \